android/app/src/main/java/com/ifabijanovic/holonet/forum/data/ForumCategoryRepository.kt

```diff
@@ -0,0 +1,114 @@
+package com.ifabijanovic.holonet.forum.data
+
+import com.ifabijanovic.holonet.app.model.Settings
+import com.ifabijanovic.holonet.forum.language.ForumLanguage
+import com.ifabijanovic.holonet.forum.model.ForumCategory
+import com.ifabijanovic.holonet.forum.model.ForumMaintenanceException
+import com.ifabijanovic.holonet.helper.StringHelper
+import io.reactivex.Observable
+import io.reactivex.schedulers.Schedulers
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+
+/**
+ * Created by feb on 20/03/2017.
+ */
+interface ForumCategoryRepository {
+    fun categories(language: ForumLanguage): Observable<List<ForumCategory>>
+    fun categories(language: ForumLanguage, parent: ForumCategory): Observable<List<ForumCategory>>
+}
+
+final class DefaultForumCategoryRepository(parser: ForumParser, settings: Settings): ForumRepositoryBase(parser, settings), ForumCategoryRepository {
+    override fun categories(language: ForumLanguage): Observable<List<ForumCategory>> {
+        val localizedSettings = this.localizedSettings(language)
+        return this.categories(localizedSettings.pathPrefix, localizedSettings.rootCategoryId)
+    }
+
+    override fun categories(language: ForumLanguage, parent: ForumCategory): Observable<List<ForumCategory>> {
+        val localizedSettings = this.localizedSettings(language)
+        return this.categories(localizedSettings.pathPrefix, parent.id)
+    }
+
+    private fun categories(language: String, categoryId: Int): Observable<List<ForumCategory>> {
+        return this.service
+                .category(language, categoryId)
+                .subscribeOn(Schedulers.io())
+                .observeOn(Schedulers.computation())
+                .unsubscribeOn(Schedulers.io())
```

Comment:
You probably do not need this `unsubscribeOn` operator if you do not have custom unsubscription logic that needs to be done on a specific thread. Like manipulating the `View`s on the main thread.